aten/src/ATen/mps/MPSAllocator.mm

```diff
@@ -561,19 +565,10 @@
   for (const auto& buffer_block : buffer_blocks) {
     // check for retain count again as the previous wait might have released the buffer
     if (buffer_block->retainCount() > 1) {
-      bool scheduledNotify = buffer_block->event->notifyEvent(
-                                /*needsLock*/ false, /*syncEvent*/ false,
-                                ^(id<MTLSharedEvent>, uint64_t) {
-                                  std::lock_guard<std::mutex> lock(m_gpu_sync_mutex);
-                                  buffer_block->gpu_sync_completed = true;
-                                  m_gpu_sync_cv.notify_one();
-                                });
-      if (scheduledNotify) {
-        std::unique_lock<std::mutex> lock(m_gpu_sync_mutex);
-        m_gpu_sync_cv.wait(lock, [&buffer_block]{ return buffer_block->gpu_sync_completed; });
-        buffer_block->gpu_sync_completed = false;
-	// after waiting, it's a good time to free some pending inactive buffers
-	freeInactiveBuffers();
+      bool waitedOnCPU = buffer_block->event->synchronize();
```

Comment:
I think this is synchronizing on CPU timeline.

Reply:
Yes, it is. We need to wait for GPU to finish before copying on to/from CPU